watcher.go

```diff
@@ -0,0 +1,67 @@
+package consul
+
+import (
+	"context"
+	"strconv"
+)
+
+type WatcherFunc func([]KeyData)
```

Comment:
Types and functions in this file would deserve a bit of documentation.

Reply:
Yep, thanks.  I've been dragging my feet on the last 20% of this PR.